documentation/src/main/asciidoc/public/reference/_search-dsl-predicate.adoc

```diff
@@ -1578,7 +1578,7 @@ While when an <<backend-elasticsearch-compatibility-opensearch,OpenSearch>> dist
 when the index is configured to have more than one shard.
 See this section of the OpenSearch link:{openSearchDocUrl}/search-plugins/knn/approximate-knn/#get-started-with-approximate-k-nn[documentation] for more details.
 
-With the Elasticsearch backend and when using <<backend-elasticsearch-compatibility-elasticsearch,Elasticsearch>>,
+With the Elasticsearch backend and when using <<backend-elasticsearch-compatibility-elasticsearch,Elasticsearch>> version < 8.12,
```

Comment:
See  that's the kind of things I'm talking about. If we keep support for KNN predicates in Elasticsearch 8.11 and below, documentation is more complex, even for people using Elasticsearch 8.12, because of these limitations. And questions from people trying to use it on 8.11... :face_exhaling: I don't want to think about that.

So yeah, I really think we should remove support for knn predicates in ES 8.11 and below. You can do it in a single commit if you want: that way, if we ever need to revert restore such support, it would be relatively straightforward.